.gitlab/build-and-test.sh

```diff
@@ -64,27 +68,59 @@ echo "~~~~~          Host: ${hostname}"
 echo "~~~~~   Project dir: ${project_dir}"
 echo "~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~"
 
-# Get the superbuild because why not.
-lbann_sb_top_dir=${build_dir}/sb
-lbann_sb_dir=${lbann_sb_top_dir}/scripts/superbuild
-mkdir -p ${lbann_sb_top_dir}
-cd ${lbann_sb_top_dir}
-
-# Sparse checkout of the SuperBuild
-git init
-git remote add origin https://github.com/llnl/lbann
-git fetch --depth=1 origin develop
-git config core.sparseCheckout true
-echo "scripts/superbuild" >> .git/info/sparse-checkout
-git pull --ff-only origin develop
-
 prefix="${project_dir}/install-deps-${CI_JOB_NAME_SLUG:-${job_unique_id}}"
 
-# Setup the build environment
-source ${project_dir}/.gitlab/setup_env.sh
+# Just for good measure...
+export CMAKE_PREFIX_PATH=${prefix}/aluminum:${prefix}/catch2:${prefix}/hwloc:${prefix}/hydrogen:${prefix}/nccl:${prefix}/spdlog:${CMAKE_PREFIX_PATH}
+
+# Allow a user to force this
+rebuild_deps=${rebuild_deps:-""}
 
-# Superbuild the dependencies (uses "${cluster}", "${prefix}", and "${lbann_sb_dir}")
+# Rebuild if the prefix doesn't exist.
 if [[ ! -d "${prefix}" ]]
+then
+    rebuild_deps=1
+fi
+
+# Rebuild if latest hashes don't match
+if [[ -z "${rebuild_deps}" ]]
+then
+    function fetch-sha {
+        # $1 is the LLNL package name (e.g., 'aluminum')
+        # $2 is the branch name (e.g., 'master')
+        curl -s -H "Accept: application/vnd.github.VERSION.sha" \
+             "https://api.github.com/repos/llnl/$1/commits/$2"
+    }
+
+    al_head=$(fetch-sha aluminum master)
+    al_latest="<not found>"
+    if [[ -f "${prefix}/al-latest.txt" ]]
+    then
+        al_latest=$(cat ${prefix}/al-latest.txt)
+    fi
+
+    h_head=$(fetch-sha elemental hydrogen)
+    h_latest="<not found>"
+    if [[ -f "${prefix}/h-latest.txt" ]]
+    then
+        h_latest=$(cat ${prefix}/h-latest.txt)
+    fi
+
+    if [[ "${al_head}" != "${al_latest}" ]]
+    then
+        echo "Aluminum hashes do not match!"
```

Comment:
In a few months, I'm probably going to forget what this indicates and worry something went wrong. Maybe "Prebuilt Aluminum hash does not match latest, rebuilding" or something.

Reply:
Updated.